pkg/services/folder/folderimpl/store.go

```diff
@@ -10,44 +10,69 @@ import (
 	"github.com/grafana/grafana/pkg/setting"
 )
 
-type store struct {
+type store interface {
+	// Create creates a folder and returns the newly-created folder.
+	Create(ctx context.Context, cmd *folder.CreateFolderCommand) (*folder.Folder, error)
+
+	// Delete deletes a folder from the folder store.
+	Delete(ctx context.Context, uid string, orgID int64) error
+
+	// Update updates the given folder's UID, Title, and Description.
+	// Use Move to change a dashboard's parent ID.
+	Update(ctx context.Context, cmd *folder.UpdateFolderCommand) (*folder.Folder, error)
+
+	// Move changes the given folder's parent folder uid and applies any necessary permissions changes.
+	Move(ctx context.Context, cmd *folder.MoveFolderCommand) (*folder.Folder, error)
+
+	// Get returns a folder.
+	Get(ctx context.Context, cmd *folder.GetFolderCommand) (*folder.Folder, error)
+
+	// GetParents returns an ordered list of parent folder of the given folder.
+	GetParents(ctx context.Context, cmd *folder.GetParentsCommand) ([]*folder.Folder, error)
+
+	// GetChildren returns the set of immediate children folders (depth=1) of the
+	// given folder.
+	GetChildren(ctx context.Context, cmd *folder.GetTreeCommand) ([]*folder.Folder, error)
+}
+
+type sqlStore struct {
 	db  db.DB
 	log log.Logger
 	cfg *setting.Cfg
 	fm  featuremgmt.FeatureManager
 }
 
-// store implements the folder.Store interface.
-var _ folder.Store = (*store)(nil)
+// sqlStore implements the store interface.
```

Comment:
```suggestion
// sqlStore implements the folder.Store interface.
```

this comment was accurate

Reply:
I just changed the store to be private